File: 2019/july2019/chef.cpp
#include<iostream>
#include<string>
#include<vector>

using namespace std;

int main(int argc, char** argv)
{
    int t;
    cin>>t;
    while(t--)
    {
        int n;
        cin>>n;
        double arr[n];
        double sum=0;
        for(int i=0;i<n;i++)
        {
            cin>>arr[i];
            sum+=arr[i];
            // cout<<arr[i]<<" ";
        }
        double mean=sum/n;
        // cout<<endl<<mean<<endl;
        for(int i=0;i<n;i++)
        {
            if(mean==arr[i])
            {
                cout<<i+1<<endl;
                break;
            }
            else if(i==n-1)
            {
                cout<<"Impossible"<<endl;
            }

        }








    }
}

File: 2019/july2019/chef.java
import java.util.*;

public class chef
{
    public static void main(String[] args)
    {
        Scanner scn=new Scanner(System.in);
        int t;
        t=scn.nextInt();
        while(t>0)
        {
            int n;
            n=scn.nextInt();
            double[] arr=new double[n];
            double sum=0;
            for(int i=0;i<n;i++)
            {
                arr[i]=scn.nextDouble();
                sum+=arr[i];
                // cout<<arr[i]<<" ";
            }
            double mean=sum/n;
            // cout<<endl<<mean<<endl;
            for(int i=0;i<n;i++)
            {
                if(mean==arr[i])
                {
                    System.out.println(i+1);
                    break;
                }
                else if(i==n-1)
                {
                    System.out.println("Impossible");
                }
    
            }
        t--;
    
        }
    scn.close();
    }
}